kratos/mpi/sources/mpi_data_communicator.cpp

```diff
@@ -474,6 +474,31 @@ bool MPIDataCommunicator::IsNullOnThisRank() const
     return mComm == MPI_COMM_NULL;
 }
 
+const DataCommunicator& MPIDataCommunicator::GetSubDataCommunicator(
+    const std::vector<int>& rRanks,
+    const std::string& rNewCommunicatorName
+    ) const
+{
+    if (ParallelEnvironment::HasDataCommunicator(rNewCommunicatorName)) {
+        const DataCommunicator& r_data_communicator = ParallelEnvironment::GetDataCommunicator(rNewCommunicatorName);
+        const int rank = Rank();
+        const auto it_find = std::find(rRanks.begin(), rRanks.end(), rank);
+        if (it_find != rRanks.end()) {
+            KRATOS_ERROR_IF_NOT(r_data_communicator.IsDefinedOnThisRank()) << "The rank " << rank << " does not participate in the existing data communicator " << rNewCommunicatorName  << " despite being in the provided rank list" << std::endl;
+            const std::size_t world_size = static_cast<std::size_t >(r_data_communicator.Size());
+            KRATOS_ERROR_IF_NOT(rRanks.size() == world_size) << "Inconsistency between the communicator world size: " << world_size << " and the number of ranks required: " << rRanks.size() << std::endl;
+            std::size_t number_of_active_ranks = 1;
+            number_of_active_ranks = r_data_communicator.SumAll(number_of_active_ranks);
+            KRATOS_ERROR_IF_NOT(number_of_active_ranks == rRanks.size()) << "Inconsistency between the number of active ranks: " << number_of_active_ranks << " and the number of ranks required: " << rRanks.size() << std::endl;
+        } else {
+            KRATOS_ERROR_IF_NOT(r_data_communicator.IsNullOnThisRank()) << "The rank " << rank << " participates in the existing data communicator " << rNewCommunicatorName << " despite not being in the provided rank list" << std::endl;
```

Comment:
I would move this check out of the condition, this should be checked in both cases

Reply:
I don't unerstand what do you mean

Reply:
this should be checked at the very top of this function

Reply:
No, because it will fail, I already tried

Reply:
then it is being used wrong IMO
This is what I meant [last week](https://github.com/KratosMultiphysics/Kratos/pull/11875#issuecomment-1840464992)

Any MPI-call on a rank where the data-comm does not have a communicator defined fails

Reply:
i would say that any **SUB**DataComm can only contain the ranks of the parent, otherwise it is not "Sub"

Reply:
I will try again, maybe I did something stupid

Reply:
Now I will try this

Reply:
Okay this is the thing, should this method be invoked from all the ranks always?, because is not actuallya fail, in the test I added it gets stuck because the subdatacommunicator is not invoked in all the ranks

Reply:
Not 100% sure either

I guess conceptually it makes sense to invoke it from all ranks, but I dont think it is possible 

I suggest you start working with it and potentially modify it in the future. This is a very niche feature, I would not be worried about API or behavior changes in the future